frontend/src/app/shared/components/dropdown/dropdown.component.ts

```diff
@@ -12,29 +12,43 @@ import { Filter } from 'src/app/models/filters/filter';
 export class DropdownComponent extends BaseComponent implements OnInit, OnDestroy {
     @Input() data: Filter[];
 
+    @Input() isSingleChoice: boolean;
+
+    type: string;
+
     @Input() btnLabel: string | undefined;
 
     @Output() selectedFilters = new EventEmitter<string[]>();
 
-    @Input() resetEvent: Observable<void>;
+    @Input() resetEvent?: Observable<void>;
 
-    @Input() selectedItems: string[];
+    @Input() selectedItems: string[] = [];
 
     public showDropdown = false;
 
     ngOnInit(): void {
-        this.resetEvent
-            .pipe(this.untilThis)
-            .subscribe(() => {
-                this.selectedItems = [];
-            });
+        if (this.resetEvent) {
+            this.resetEvent
+                .pipe(this.untilThis)
+                .subscribe(() => {
+                    this.selectedItems = [];
+                });
+        }
+
+        this.type = this.isSingleChoice ? 'radio' : 'checkbox';
```

Comment:
why do we need this flag?

Reply:
if we have an ability to select only one option, than we have drop-down menu with radio buttons. In other case we have checkbox

Reply:
in this case, I belive we can redactor lines 46-51 and combine them
and in case we have only two cases it can be rewritten to if/else